src/functions.js

```diff
@@ -133,7 +154,29 @@ var _ = Mavo.Functions = {
 			return n && n > 1? [arr] : arr;
 		}
 
-		return n && n > 1? arr.slice(-n) : arr[arr.length - 1];
+		if (n && n > 1) {
+			var ret = [];
+
+			for (i = arr.length-1; i>=0 && ret.length<n; i--) {
+
+				if (Mavo.value(arr[i]) !== null) {
+					ret.push(arr[i]);
+				}
+
+			}
+
+			return ret;
+		} 
+		else {
+			var i = arr.length - 1;
+
+			while (i>=0 && Mavo.value(arr[i]) == null) {
+				i--;
```

Comment:
This can also be turned into a for loop. It's a little less typical than the first one, but still a for loop would be more readable. If you disagree, we can discuss in person.